common/analysis/lint_rule_status.cc

```diff
@@ -58,6 +58,42 @@ void LintStatusFormatter::FormatLintRuleStatus(std::ostream* stream,
   }
 }
 
+struct LintViolationWithStatus {
+  const LintViolation *violation;
+  const LintRuleStatus *status;
+
+  LintViolationWithStatus(const LintViolation *v, const LintRuleStatus *s)
+      : violation(v), status(s) {}
+
+  bool operator<(const LintViolationWithStatus& r) const {
+    return (violation->token.text.data() < r.violation->token.text.data());
```

Comment:
// compares addresses which correspond to locations within the same string

Reply:
sure, added the comment